examples/shared/src/main/scala/decode.scala

```diff
@@ -10,12 +10,10 @@ trait Decoder[T] { def decode(str: String): T }
 object Decoder {
 
   /** decodes strings */
-  implicit val string: Decoder[String] = new Decoder[String] {
-    def decode(str: String): String = str
-  }
+  implicit val string: Decoder[String] = (str: String) => str
 
   /** decodes ints */
-  implicit val int: Decoder[Int] = new Decoder[Int] { def decode(str: String): Int = str.toInt }
+  implicit val int: Decoder[Int] = (str: String) => str.toInt
```

Comment:
Do you actually need the `: String` type annotation? I think `_.toInt` should just work.

Reply:
Good catch, I was being cautious after `identity` failed previously